meilisearch-core/src/update/documents_addition.rs

```diff
@@ -233,6 +243,11 @@ pub fn apply_addition<'a, 'b>(
 
     index.main.put_schema(writer, &schema)?;
 
+    let new_external_docids = fst::Map::from_iter(new_external_docids.iter().map(|(u, i)| (u, *i as u64)))?;
```

Comment:
it is not clear what `u` and `i` are, even more now that you changed userid to external_docid